### DISCUSSION TITLE
Zephyr vs. RIOT comparison

### DISCUSSION THREAD

**Discussion by luizvilla** [zephyrproject-rtos/zephyr#32870] [Q&A] [answered]
Hello everyone, 

I'm Luiz Villa, a researcher at the University of Toulouse. I'm working with software defined power converters and my team is looking into using an RTOS to manage real-time micro-controllers. 
We ran a benchmark between Zephyr and RIOT OS that we would like to share with you. 
**Spoiler: Zephyr performed poorly.** 
Here's the summary: 

![image](https://user-images.githubusercontent.com/25325731/109948090-46f35900-7cda-11eb-8b4f-f1c0512d38a7.png)
Our test code is available at https://gitlab.laas.fr/owntech/zephyr/-/tree/test_adc_g4

Our target : Nucleo-G474RE

Our main conclusions:

- Zephyr has an overhead of almost 27us compared with 10us from RIOT
- Zephyr consumes 34% of CPU cycles compared with 14% from RIOT
- Zephyr has a period of 12us when we flood the shell compared with 1us from RIOT
- Zephyr has a minimum thread period with an active shell of 38us compared with 40us from RIOT

Our questions for the community:

- Does anyone has the same issues? 
- Have you managed to bring the execution time of Zephyr below this threshold? 
- Are we missing something on our code?

**Comment by nashif** [comment, marked as answer]
discussion happening here: https://github.com/zephyrproject-rtos/zephyr/issues/32875
